### DISCUSSION TITLE
Node Module Support for Flows

### DISCUSSION THREAD

**Discussion by Kopf02** [directus/directus#15886] [Feature Requests]
According to VM2 it is possible to add a require section to the VM options, which allows defining modules, that should be allowed to require in the VM.
Since (at least) the Execute Script Operation uses the VM2 package it would be awesome to allow certain modules to be allowed. That would give much more flexibility for flows.

**Comment by br41nslug** [comment]
You can manage which modules are allowed in VM2 using the following environment variable `FLOWS_EXEC_ALLOWED_MODULES` which accepts an array of modules to allow

**Reply by raphourbe** [reply]
I tried it with the stripe module and it works perfectly.

`FLOWS_EXEC_ALLOWED_MODULES=stripe,` in the .env file of directus.

**Reply by MentalGear** [reply]
Would be great if this setting (node.js for Flows) was also available in the cloud version.

**Reply by br41nslug** [reply]
We're actively working on making (at least parts of this) available on cloud but as you might imagine there are some security considerations when running untrusted code on a shared infrastructure.

**Comment by dgbau** [comment]
How would this feature best be used within a docker environment, just work an `npm install x-package` command into the Dockerfile for the Directus container upon creation?

**Reply by rijkvanzanten** [reply]
That'd be my suggestion, yeah!
